tests/integration/modifiers/before-unload-test.ts

```diff
@@ -0,0 +1,24 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { TestContext } from 'ember-test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import sinon from 'sinon';
+
+interface ModifierTestContext extends TestContext {
+    listener: any;
+}
+
+module('Integration | Modifier | before-unload', hooks => {
+    setupRenderingTest(hooks);
+
+    test('it adds event listener', async function(this: ModifierTestContext, assert) {
+        this.listener = sinon.fake();
+
+        await render(hbs`<div {{before-unload this.listener}}></div>`);
+        sinon.assert.notCalled(this.listener);
+        window.dispatchEvent(new Event('beforeunload'));
+        sinon.assert.calledOnce(this.listener);
+        assert.ok(true, 'beforeunload listener called');
+    });
```

Comment:
This could potentially use a couple more test cases, such as one to remove an event listener if it already has one, and the other to remove a listener on `willRemove()`.

Reply:
It might be nice if you could test some of the other .ts code while you're here, looking at the coveralls report. I'm not sure how tricky that will be with the nature of the code.